anitya/lib/backends/github.py

```diff
@@ -59,18 +65,168 @@ def get_versions(cls, project):
             when the versions cannot be retrieved correctly
 
         '''
+        owner = None
+        repo = None
         if project.version_url:
-            url_template = 'https://github.com/%(version_url)s/tags'
-            version_url = project.version_url.replace(
-                'https://github.com/', '')
-            url = url_template % {'version_url': version_url}
+            url = project.version_url
         elif project.homepage.startswith('https://github.com'):
-            url = project.homepage
+            url = project.homepage.replace('https://github.com/', '')
             if url.endswith('/'):
-                url = project.homepage[:-1]
-            url += '/tags'
+                url = url[:-1]
         else:
             raise AnityaPluginException(
                 'Project %s was incorrectly set-up' % project.name)
 
-        return get_versions_by_regex(url, REGEX, project)
+        if url:
+            (owner, repo) = url.split('/')
+
+        if (not owner) or (not repo):
```

Comment:
It's exactly equivalent, but ``if not (owner and repo)`` might read a little cleaner.

Also, I know this is just a refactor of what was there before, but it makes more sense to validate this *before* it goes in the database, rather than right here. It'd tell the user up-front they've put in incorrect data. Might be a nice refactor for a future PR.

Reply:
Yes, this should be probably validated on frontend in javascript, but you also need to validate it before saving it to db (in case somebody is trying something nasty). This could be different for every backend, so it will be good to have validation method as part of backend and the db method could call it before inserting the data.

What do you think?